### PR TITLE
Add AArch64 variant to Artifact Bundle

### PR DESCRIPTION
**Pull request by csjones**
Resolves https://github.com/nicklockwood/SwiftFormat/issues/1921

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -36,10 +36,33 @@ jobs:
...
 
+  linux_aarch64:
+    name: Build SwiftFormat for Linux AArch64
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Static Linux SDKs
+        run: |
+          swift sdk install \
```
Comment (benedek-kozma-zenitech) [inline .github/workflows/release.yml:47]: The static sdk needs to match the version of Swift and there is nothing here that ensures this, so it's better to use Dockerfile for building and exporting the binary: https://docs.docker.com/build/building/export/
The x64 version also needs to be built with the static SDK, currently it most likely crashes on OSs other than Ubuntu 20.04 due to different dependency versions.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -36,10 +36,33 @@ jobs:
...
+            https://download.swift.org/swift-6.0.2-release/static-sdk/swift-6.0.2-RELEASE/swift-6.0.2-RELEASE_static-linux-0.0.1.artifactbundle.tar.gz \
+            --checksum aa5515476a403797223fc2aad4ca0c3bf83995d5427fb297cab1d93c68cee075
+      - name: Build aarch64
+        run: |
+          swift build --configuration release --enable-dead-strip --static-swift-stdlib --swift-sdk aarch64-swift-linux-musl
```
Comment (benedek-kozma-zenitech) [inline .github/workflows/release.yml:52]: `--static-swift-stdlib` is not needed
I had some time to think and in my opinion, `-Xlinker -s` should probably be added to the build arguments in `Scripts/build-linux-release.sh` to reduce size.

Comment (benedek-kozma-zenitech) [reply .github/workflows/release.yml:52]: (my personal account is @Cyberbeni , I made the update to the Dockerfile)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -28,22 +28,58 @@ jobs:
...
+          strip artifacts/workspace/swiftformat
+          mv artifacts/workspace/swiftformat "${HOME}/swiftformat_linux"
+      - name: Upload Artifact
         uses: actions/upload-artifact@v4
         with:
           name: swiftformat_linux
           path: ~/swiftformat_linux
           retention-days: 5
 
+  linux_aarch64:
```
Comment (benedek-kozma-zenitech) [inline .github/workflows/release.yml:51]: This job is missing the 'Set Version Number' step.

We could also build both architectures in one job: `platforms: linux/amd64,linux/arm64` (I suspect downloading the docker base image and the static sdk are taking the longest). Then the output will be in `artifacts/linux_amd64` and `artifacts/linux_arm64`.

**Review by benedek-kozma-zenitech — APPROVED** [review]

**Comment by nicklockwood** [comment]
@csjones @benedek-kozma-zenitech I've noticed that the macos binary in the Artifact Bundle is larger than the one I ship with the release. Should we maybe be including ` -Xswiftc -O` in the arguments to `swift build`?

**Pull request merged**